state/workingset.go

```diff
@@ -63,6 +63,18 @@ type (
 	}
 )
 
+// Context provides the runactions with auxiliary information.
+type Context struct {
+	// producer who compose those actions
+	ProducerAddr string
+
+	// gas Limit for perform those actions
+	GasLimit *uint64
```

Comment:
Why do we still need GasLimit in context info?

Reply:
I revisit the logic, seems that it's better to use gasLimit in runactions. So that it will be consistent with run executions.  Since proposer will also run actions, it's better we handle gasLimit inside run actions.